### DISCUSSION TITLE
save card for future use

### DISCUSSION THREAD

**Discussion by shereefhamed** [flutter-stripe/flutter_stripe#825] [Q&A]
I want to use PaymentSheet
first I create intent payment by this code:
 Map<String, dynamic> body = {
        'amount': amount,
        'currency': currency,
        'payment_method_types[]': 'card',
        'setup_future_usage': 'off_session',
      };
      final url = Uri.parse(StripeService.paymentApiUrl);
      var response = await http.post(
        url,
        body: body,
        headers: StripeService.headers,
      );
then I use client secret to display payment sheet

await Stripe.instance.initPaymentSheet(
          paymentSheetParameters: SetupPaymentSheetParameters(
        paymentIntentClientSecret: paymentIntent['client_secret'],
        merchantDisplayName: 'My name',
        applePay: true,
        googlePay: true,
        billingDetails: billingDetails,
        merchantCountryCode: 'EG',
      ));
      await Stripe.instance.presentPaymentSheet();
how can I save card info after payment completed

**Comment by AristideVB** [comment]
Is it possible to save payment methods & retrieve them for future use ? I'd like to create an onboarding UI in my app where the customer payment card details are saved while not processing any payment

**Comment by Kazutaka333** [comment]
I assume we need SetupIntent instead of PaymentIntent and initialize payment sheet with setupintent but I haven't tried it yet
https://stripe.com/docs/payments/save-and-reuse

**Reply by SogoGolf** [reply]
did you try this with success ? if so can u help by providing detailed steps using this flutter_stripe lib, as we are having a lot of trouble getting this to work (save card details on a Customer to use later)

**Reply by Kazutaka333** [reply]
Yeah it worked.

**Reply by Kazutaka333** [reply]
Reading relevant parts of stripe official documentation helped me a lot if you haven't done so. This flutter library is architected just to use iOS/Android SDK, so the backend work you need to be done is the exact same as described in official iOS/Android doc.

**Reply by kean-fetchpet** [reply]
@Kazutaka333 did it work for apple pay and google pay as well? Did you use payment sheet but with setup intent secret? Thanks.
